comp/netflow/server/integration_test.go

```diff
@@ -3,60 +3,57 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2022-present Datadog, Inc.
 
-package netflow
+//go:build test
+
+package server
 
 import (
-	"fmt"
-	"strings"
 	"testing"
 	"time"
 
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/comp/core/log"
-	"github.com/DataDog/datadog-agent/pkg/aggregator"
-	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/epforwarder"
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+	nfconfig "github.com/DataDog/datadog-agent/pkg/netflow/config"
 	"github.com/DataDog/datadog-agent/pkg/netflow/flowaggregator"
 	"github.com/DataDog/datadog-agent/pkg/netflow/testutil"
 	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
+func singleListenerConfig(flowType common.FlowType, port uint16) *nfconfig.NetflowConfig {
```

Comment:
Shall we pass yaml config

```
network_devices:
  netflow:
    enabled: true
    aggregator_flush_interval: 1
    listeners:
      - flow_type: netflow5 # netflow, sflow, ipfix
        bind_host: 127.0.0.1
        port: %d # default 2055 for netflow
```

as input of this integration test instead of `NetflowConfig` to have better coverage? By reading yaml via nfconfig.ReadConfig.

I know the NetflowConfig is already unit test, but using the yaml in the integration test will provide additional coverage.

Reply:
The downside to using yaml is that it's a lot easier to break things - if you get the indentation wrong, for example, the config parser won't complain, it'll just silently populate the configuration with defaults. This seems like it would make it very easy to accidentally test the wrong thing.